prowler/providers/azure/azure_provider.py

```diff
@@ -244,14 +380,86 @@ def setup_session(
 
         return credentials
 
-    def __check_service_principal_creds_env_vars__(self):
+    @staticmethod
+    def test_connection(
+        az_cli_auth,
+        sp_env_auth,
+        browser_auth,
+        managed_identity_auth,
+        tenant_id,
+        region,
+        credentials=None,
+    ) -> tuple[DefaultAzureCredential, AzureRegionConfig]:
+        """
+        Test the connection to an Azure subscription using the provided credentials.
+
+        Args:
+            credentials: The credentials object used to authenticate with Azure.
+            az_cli_auth (bool): Flag indicating whether to use Azure CLI authentication.
+            sp_env_auth (bool): Flag indicating whether to use Service Principal authentication with environment variables.
+            browser_auth (bool): Flag indicating whether to use interactive browser authentication.
+            managed_identity_auth (bool): Flag indicating whether to use managed identity authentication.
+            tenant_id (str): The Azure Active Directory tenant ID.
+            region (str): The Azure region.
+
+        Returns:
+            tuple: A tuple containing the credentials and region configuration objects.
+        """
+        try:
+            region_config = AzureProvider.setup_region_config(
+                validate_azure_region(region)
```

Comment:
If the validate raises an exception please catch it in this method and raise a `validation_error`

Reply:
Done